### DISCUSSION TITLE
CosmosDbContainer on GitHub Actions stuck

### DISCUSSION THREAD

**Discussion by mammoetsaus** [testcontainers/testcontainers-dotnet#1042] [Q&A] [answered]
Hi all,

When running my CosmosDb testcontainer on GitHub actions, the workflow halts to complete the readiness checks.
I am using _ubuntu-latest_ as job runner, have Testcontainers.CosmosDb 3.5.0 installed, and have configured my container as follows:

![image](https://github.com/testcontainers/testcontainers-dotnet/assets/5879397/c373dc8a-3e61-457c-859f-008d05c4ac1d)

I face the same issues with either no wait strategy or other strategies (e.g., HTTP Request, Command Done, Container Healthy).

This is part of my workflow output:
![image](https://github.com/testcontainers/testcontainers-dotnet/assets/5879397/60bd0820-409e-4407-ac59-df912759e420)

Does anyone have suggestions on how to resolves this?

Thanks in advance!

**Comment by HofmeisterAn** [comment, marked as answer]
I am sorry, but the Cosmos DB Linux Emulator Docker image does not run on Microsoft's CI environment (GitHub and Azure DevOps). Depending on the configuration, it may run on private runners (agents). It looks like Microsoft is currently looking into this issue. You can find more information here: https://github.com/Azure/azure-cosmos-db-emulator-docker/issues/45.

**Reply by mammoetsaus** [reply]
Thanks for the swift reply, I'll look into running a self-hosted agent with different chipset for now 😊

**Comment by ishaan-kapoor** [comment]
Has this issue been solved yet?
@mammoetsaus , please share the workaround you used.

**Reply by NelsonBN** [reply]
There is light at the end of the tunnel and Microsoft is working on a [Linux-based image](https://learn.microsoft.com/en-gb/azure/cosmos-db/emulator-linux) version that already runs in [CI/CD pipelines](https://github.com/NelsonBN/demo-dotnet-integration-tests-cosmosdb-container/actions). 

Additionally, some work is already underway for Testcontainers to support the new version of these images.
- [Open Discussion](https://github.com/testcontainers/testcontainers-dotnet/discussions/1306)
- [Possible PR](https://github.com/testcontainers/testcontainers-dotnet/pull/1324)
